core/src/main/java/org/elasticsearch/search/aggregations/pipeline/movavg/MovAvgPipelineAggregator.java

```diff
@@ -193,6 +201,60 @@ public InternalAggregation reduce(InternalAggregation aggregation, ReduceContext
         return factory.create(newBuckets, histo);
     }
 
+    private double[] minimize(List<? extends InternalHistogram.Bucket> buckets, InternalHistogram histo, double[] modelCoef) {
```

Comment:
I'm a bit torn with this. To me it feels like this might be a function of the model rather than the pipeline aggregator. It might be better in the MovAvgModel abstract class? It does then give us the ability to have different minimisers for different models if we find it appropriate in the future. What do you think?